cnf-certification-test/lifecycle/suite.go

```diff
@@ -141,3 +149,93 @@ func testPodsOwnerReference(env *provider.TestEnvironment) {
 		gomega.Expect(0).To(gomega.Equal(len(badPods)))
 	})
 }
+
+func testScaling(env *provider.TestEnvironment) {
+	ocpClient := ocpclient.NewOcpClient()
+	testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestDeploymentScalingIdentifier)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		ginkgo.By("Testing deployment scaling")
+		defer env.SetNeedsRefresh()
+
+		if len(env.Deployments) == 0 {
+			ginkgo.Skip("No test deployments found.")
+		}
+		failedDeployments := []string{}
+		skippedDeployments := []string{}
+		for i := range env.Deployments {
+			//runScalingfunc(env.Deployments[i])
+			//defer restoreDeployments(env.Deployments[i])
+			skipped, failed := testScaleDeployment(ocpClient.AppsClients, env.Deployments[i], env.HorizentalScaller)
+			if skipped {
+				skippedDeployments = append(skippedDeployments, env.Deployments[i].Name)
+			}
+			if failed {
+				failedDeployments = append(failedDeployments, env.Deployments[i].Name)
+			}
+		}
+
+		if len(skippedDeployments) > 0 {
+			tnf.ClaimFilePrintf("not ready deployments : %v", skippedDeployments)
+		}
+		if len(failedDeployments) > 0 {
+			tnf.ClaimFilePrintf(" failed deployments: %v", failedDeployments)
+		}
+		gomega.Expect(0).To(gomega.Equal(len(failedDeployments)))
+		gomega.Expect(0).To(gomega.Equal(len(skippedDeployments)))
+	})
+}
+
+func testScaleDeployment(oc *appv1client.AppsV1Client, deployment *v1app.Deployment, hpas map[string]*v2.HorizontalPodAutoscaler) (skipped, failed bool) {
+	ready := false
+	//ocp := ocpclient.NewOcpClient()
+	for _, condition := range deployment.Status.Conditions {
+		if condition.Type == v1app.DeploymentAvailable {
+			ready = true
+			break
+		}
+	}
+	if !ready {
+		return true, true
+	}
+	key := deployment.Namespace + deployment.Name
+	if hpa, ok := hpas[key]; ok {
+		//spec := hpa.Spec
+		var min int32
+		var max int32
+
+		if hpa.Spec.MinReplicas != nil {
+			min = *hpa.Spec.MinReplicas
+		} else {
+			min = 1
+		}
+		max = hpa.Spec.MaxReplicas
+
+		// scale up the deployment
+		min++
+		max++
+		hpa.Spec.MinReplicas = &min
+		hpa.Spec.MaxReplicas = max
```

Comment:
Why is `min` a pointer and `max` is not?

Reply:
min is a pointer, so if it's nil,  k8s will use the default value which is 0. having a pointers allows k8s to know if the user provided a value or not.
Not sure why max is not a pointer, maybe should be always more than 1 ? 🤔